deposit/src/main/java/edu/unc/lib/deposit/normalize/BagIt2N3BagJob.java

```diff
@@ -92,10 +92,14 @@ public void runJob() {
             MandatoryVerifier.checkBagitFileExists(bagReader.getRootDir(), bagReader.getVersion());
 
             try (BagVerifier verifier = new BagVerifier()) {
+                interruptJobIfStopped();
                 verifier.isComplete(bagReader, false);
+                interruptJobIfStopped();
                 verifier.isValid(bagReader, false);
             }
 
+            interruptJobIfStopped();
```

Comment:
Would this cause any noticeable slowdown in ingest, checking for interruptions so often in succession?

Reply:
I tried to balance that by not adding too many of these checks in, and not adding them at all for some of the jobs that finish quickly. I decided which jobs to add it to by looking at the runtimes for each job in previous testing tickets. These checks should be pretty fast since redis is on the same server, and is generally very fast. But it is still a new cost.

In this particular case, there is a max of 3 calls for the job, unlike some of the others where it depends on the number of objects, so I'm not too worried about it in this case. I tried to place them in between long running steps, before it starts writing to the deposit model.